src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_sysfs.c

```diff
@@ -513,11 +524,20 @@ static ssize_t ready_show(struct device *dev,
 	else {
 		/*
 		 * If xocl and xclmgmt are not in the same daemon,
-		 * mark the card as ready only when both MB channel
-		 * and daemon are ready
+		 * mark the card as ready when
+		 *  1. both MB channel and daemon are ready
+		 *  This is for case cloud vendor controls the xclbin download,
+		 *  like azure, aws F1
+		 *  2. MB channel is ready
+		 *     and
+		 *     XCL_MAILBOX_REQ_LOAD_XCLBIN is disabled
```

Comment:
wow... "ready" state is a very generic state for a device.
but, here, we use disable state of one specific mailbox op code to determine the generic "ready" state of the board. this sounds very hacky to me. it feels that we are missing one abstract layer here...
what if admin disabled another op code? what if this XCL_MAILBOX_REQ_LOAD_XCLBIN op code is removed completely in future release?
i'm wondering if we can introduce another flag for mgmt to explicitly tell user that there will be no daemon required?

Reply:
changed as discussed